File: advanced_javascript/class/constructor/constructor.py
class Lion:
    def __init__(self, name):
        self.name = name
        # self는 전달된 객체
        # self.name은 인스턴스 변수

    def getName(self):
        print("내 이름은 " + self.name)


myLion = Lion("King")
myLion.getName()

File: advanced_javascript/class/constructor/constructor.js
class Lion {
  constructor(name, age) {
    this.name = name;
    this.age = age;
    //생성자 함수 내의 this 는
    //new를 통해 만들어진 새로운 변수가 된다.
  }
  getName() {
    console.log("내 이름은 " + this.name);
  }
  getAge() {
    console.log(`나는 ${this.age}살이얌!`);
  }
}

myLion = new Lion("King", 22); //인스턴스 생성
myLion.getName();
myLion.getAge();
